### ISSUE TITLE
Error while generating package metadata

### ISSUE THREAD

**Issue by ptfmiller** [zeldamods/oead#33]
While I try to pip install bcml, here is what I get: 

```Collecting oead~=1.2.4 (from bcml)
  Using cached oead-1.2.7.tar.gz (4.2 MB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [18 lines of output]
      /private/var/folders/p0/83wz866d50xfhv49j51sfd6r0000gn/T/pip-install-8uo0j8yo/oead_7a24d08f414e47bbbc1c104cd092ebca/versioneer.py:421: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/private/var/folders/p0/83wz866d50xfhv49j51sfd6r0000gn/T/pip-install-8uo0j8yo/oead_7a24d08f414e47bbbc1c104cd092ebca/setup.py", line 66, in <module>
          version=versioneer.get_version(),
                  ^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/p0/83wz866d50xfhv49j51sfd6r0000gn/T/pip-install-8uo0j8yo/oead_7a24d08f414e47bbbc1c104cd092ebca/versioneer.py", line 1480, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "/private/var/folders/p0/83wz866d50xfhv49j51sfd6r0000gn/T/pip-install-8uo0j8yo/oead_7a24d08f414e47bbbc1c104cd092ebca/versioneer.py", line 1412, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/p0/83wz866d50xfhv49j51sfd6r0000gn/T/pip-install-8uo0j8yo/oead_7a24d08f414e47bbbc1c104cd092ebca/versioneer.py", line 342, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
```

From another repo, it looks like `versioneer` is deprecated.

**Comment by davidzchen** [comment]
I have the same error. Here is my error output:

```
PIP STDOUT
----------
Collecting oead
  Using cached oead-1.2.7.tar.gz (4.2 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'

PIP STDERR
----------
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [31 lines of output]
      /private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-install-gwgoab9c/oead_ad354b113b2d4de491695a7ea8c19b27/versioneer.py:421: SyntaxWarning: invalid escape sequence '\s'
        LONG_VERSION_PY['git'] = '''
      Traceback (most recent call last):
        File "/Users/dzc/.local/pipx/shared/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/dzc/.local/pipx/shared/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/dzc/.local/pipx/shared/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-build-env-qi36tzzd/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-build-env-qi36tzzd/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-build-env-qi36tzzd/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-build-env-qi36tzzd/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 66, in <module>
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-install-gwgoab9c/oead_ad354b113b2d4de491695a7ea8c19b27/versioneer.py", line 1480, in get_version
          return get_versions()["version"]
                 ^^^^^^^^^^^^^^
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-install-gwgoab9c/oead_ad354b113b2d4de491695a7ea8c19b27/versioneer.py", line 1412, in get_versions
          cfg = get_config_from_root(root)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/d3/yj9wzlg96ql05cjzm329d76r0000gn/T/pip-install-gwgoab9c/oead_ad354b113b2d4de491695a7ea8c19b27/versioneer.py", line 342, in get_config_from_root
          parser = configparser.SafeConfigParser()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'configparser' has no attribute 'SafeConfigParser'. Did you mean: 'RawConfigParser'?
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Can you please update versioneer?

I am on macOS Sonoma 14.4.1 on Apple Silicon. Python version:

```
$ python3 --version
Python 3.12.3
```

**Comment by leoetlino** [comment]
This should be fixed by #35. I'm releasing a new version as I speak - v1.2.9 will be available very soon.

**Comment by davidzchen** [comment]
Thanks, Léo!
